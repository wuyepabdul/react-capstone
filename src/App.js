import React from 'react';
import { BrowserRouter, Route, Routes } from 'react-router-dom';
import NavBar from './components/NavBar';
import CountriesList from './pages/CountriesList';
import CountryDetails from './pages/CountryDetails';

const App = () => (
  <BrowserRouter>
    <NavBar />
    <Routes>
      <Route path="/" element={<CountriesList />} />
      <Route path="/country/:countryName" element={<CountryDetails />} />
    </Routes>
  </BrowserRouter>
);

export default App;
