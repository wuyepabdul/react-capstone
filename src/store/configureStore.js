import { createStore, applyMiddleware, combineReducers } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import { composeWithDevTools } from 'redux-devtools-extension';
import countriesListReducer from './countries/countriesList';
import countryReducer from './country/country';
import regionReducer from './region/region';

const reducers = combineReducers({
  countriesList: countriesListReducer,
  countryDetails: countryReducer,
  regionDetails: regionReducer,
});

const middleWares = [thunk, logger];

const configureStore = createStore(
  reducers,
  composeWithDevTools(applyMiddleware(...middleWares)),
);

export default configureStore;
