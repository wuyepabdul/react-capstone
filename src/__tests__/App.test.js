import '@testing-library/jest-dom/extend-expect';
import { render, screen, cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import App from '../App';

describe('AppRouting and Navbar', () => {
  afterEach(() => {
    cleanup();
  });
  const initialState = {
    countriesList: [],
    countryDetails: {},
  };
  const mockStore = configureStore();
  const store = mockStore(initialState);

  test('navbar is loaded and contains elements', () => {
    render(<Provider store={store}><App /></Provider>);
    const navBar = screen.getByTestId('navigation-bar');
    expect(navBar).toBeInTheDocument();
    expect(navBar.childNodes[0].firstChild.textContent).toBe('Covid-19 Reports');
    expect(navBar.childNodes[1].textContent).toBe('March, 2020');
    expect(navBar.childNodes.length).toEqual(2);
  });
  test('HomePage/CountriesList Component is loaded', () => {
    render(<Provider store={store}><App /></Provider>);
    const countriesListComponent = screen.getByTestId('home-page');
    expect(countriesListComponent).toBeInTheDocument();
  });
  test('snapshot of Whole APP', () => {
    const mainAppComponent = render(<Provider store={store}><App /></Provider>);
    expect(mainAppComponent).toMatchSnapshot();
  });
});
