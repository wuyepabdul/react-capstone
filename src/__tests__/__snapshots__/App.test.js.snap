// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouting and Navbar snapshot of Whole APP 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navbar"
        data-testid="navigation-bar"
      >
        <header
          class="logo-container"
        >
          <h1
            class="title"
          >
            Covid-19 Reports
          </h1>
        </header>
        <p>
          March, 2020
        </p>
      </nav>
      <div
        class="countries-container"
        data-testid="home-page"
      />
    </div>
  </body>,
  "container": <div>
    <nav
      class="navbar"
      data-testid="navigation-bar"
    >
      <header
        class="logo-container"
      >
        <h1
          class="title"
        >
          Covid-19 Reports
        </h1>
      </header>
      <p>
        March, 2020
      </p>
    </nav>
    <div
      class="countries-container"
      data-testid="home-page"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
